Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -331,17 +334,31 @@ public class Klaviyo : NSObject {
      - Returns: Void
      */
     public func addPushDeviceToken(deviceToken: Data) {
-        let apnDeviceToken = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
+        apnDeviceToken = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
         let personInfoDictionary : NSMutableDictionary = NSMutableDictionary()
         personInfoDictionary[CustomerPropertiesAppendDictKey] = [CustomerPropertiesAPNTokensDictKey: apnDeviceToken]
         trackPersonWithInfo(personDictionary: personInfoDictionary)
         
         let defaults = UserDefaults.standard
         defaults.setValue(apnDeviceToken, forKey: CustomerPropertiesAPNTokensDictKey)
     }
-    
-    
-    
+
+    /**
+     getPushDeviceToken: Get push device token if registered
+     Provides managed access to the in-memory apnDeviceToken property,
+     to fetch from UserDefaults if it hasn't been yet
+
+     - Returns: String?
+     */
+    public func getPushDeviceToken() -> String? {
+        if apnDeviceToken == nil {
```

Comment:
💭  Does it make sense to keep `UserDefaults` the source of truth for getting the device token instead of the in memory variable just so that we don't have to share the variable between functions here?

Reply:
If we did this this function could be a simple read only computed property. Just a suggestion.

Reply:
Just a little quicker to have it in memory too is all